backend/TaskTree/Controllers/TasksController.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using TaskTree.Models;
+using TaskTree.Models.Requests;
+using TaskTree.Models.Responses;
+using Task = TaskTree.Models.Task;
+
+namespace TaskTree.Controllers
```

Comment:
Every authorized route (POST, PUT, and the GETs), should have the [Authorize] attribute above it, as well as the `CurrentUserIdDoesNotMatch()` check. for the GET Tasks endpoint, you'll probably have to get the project and then use the userId from that to call `CurrentUserIdDoesNotMatch`